src/com/sap/piper/JenkinsController.groovy

```diff
@@ -0,0 +1,75 @@
+package com.sap.piper
+
+class JenkinsController implements Serializable {
+    def script
+    def timeout = 3600
+    String jenkinsUrl = "http://localhost:8080"
+
+    def createJob(String gitUri, String jobName) {
+        String genericJobTemplate = script.libraryResource "jobTemplate.xml"
+        String jobXml = Utils.fillTemplate(genericJobTemplate, [guid: UUID.randomUUID().toString(), gitUri: gitUri])
+        return this.createJobFromJobXml(jobXml, jobName)
+    }
+
+    def waitForJenkinsStarted() {
+        def timeout = 120
+        def timePerLoop = 5
+
+        for (int i = 0; i < timeout; i += timePerLoop) {
+            script.sleep timePerLoop
+            try {
+                if (retrieveJenkinsStatus() == 'NORMAL') {
+                    return true
+                }
+            } catch (Exception e) {
+                script.echo "Could not retrieve status for Jenkins. Retrying..."
+                script.echo e.getMessage()
+                continue
+            }
+            return false
+        }
+        script.error("Timeout: Build of waiting for Jenkins status}")
+    }
+
+    JenkinsJobController createJobFromJobXml(String jobXml, String jobName) {
+        def retry = 0
+        def jobSuccessfullyCreated = false
+
+        def createJobEndpoint = BashUtils.quoteAndEscape("${jenkinsUrl}/createItem?name=${jobName}")
+
+
+        jobXml = BashUtils.quoteAndEscape(jobXml)
+
+        while (!jobSuccessfullyCreated && retry < 5) {
+            script.dir('resources') {
+                script.sh "curl -X POST ${createJobEndpoint} -v --header \"Content-Type: application/xml\" -d ${jobXml}"
+                script.sleep 5
+                jobSuccessfullyCreated = this.isJobAvailable(jobName)
+                retry++
+            }
+            if (jobSuccessfullyCreated) {
+                script.echo("Job has been successfully imported to test server")
+                return new JenkinsJobController(script: script, jenkinsUrl: jenkinsUrl, jobName: jobName)
+            }
+        }
+        script.error("Job has not been successfully imported to test server")
+    }
+
+    def isJobAvailable(String jobName) {
+        def jobStateEndpoint = BashUtils.quoteAndEscape("${jenkinsUrl}/job/${jobName}/api/json")
+        def statusCode = script.sh(
+            script: "curl -XGET -s -o /dev/null -I -w \"%{http_code}\" ${jobStateEndpoint}",
```

Comment:
Is it possible to check that in a more directy way than curl pointing to the server itself?

E.g.

pseudoCode: ```Jenkins.instance.getAllItems(Job.class).contains(jobName)```


This comment is void in case we try to create a job on another jenkins running on the same machine (`localhost`), but on another port.